src/dbus/mpris.rs

```diff
@@ -314,9 +314,14 @@ impl SpotMprisPlayer {
 
     #[dbus_interface(property)]
     pub fn volume(&self) -> f64 {
-        0f64
+        self.state.volume()
     }
 
     #[dbus_interface(property)]
-    pub fn set_volume(&self, value: f64) {}
+    pub fn set_volume(&self, value: f64) -> Result<()> {
+        self.state.set_volume(value);
```

Comment:
right now its not a big deal at all, but in the long term I´d recommend not modifying the state here, but only when the change has been acknowledged by the main process (written to the app state). this keeps all mutations of the state (and dbus outbound notifications) in one place, whether this happened through mpris or through something else (like a slider) :)

Reply:
I agree! How would I get the volume from the main process from mpris?

Reply:
my files are really poorly organized but you can see that in the `impl EventListener for AppPlaybackStateListener`

Reply:
Nice! I'll be comitting my changes soon but I'll take some days to respond since it's new years